components/feature/pwa/src/main/java/mozilla/components/feature/pwa/ManifestStorage.kt

```diff
@@ -25,10 +25,24 @@ class ManifestStorage(context: Context) {
      * Load a Web App Manifest for the given URL from disk.
      * If no manifest is found, null is returned.
      *
-     * @param startUrl URL of site. Should correspond to manifest's start_url.
+     * @param url URL of site. Should correspond to manifest's start_url.
      */
-    suspend fun loadManifest(startUrl: String): WebAppManifest? = withContext(Dispatchers.IO) {
-        manifestDao.value.getManifest(startUrl)?.manifest
+    suspend fun loadManifest(
+        url: String,
+        allowTimedout: Boolean = false
+    ): WebAppManifest? = withContext(Dispatchers.IO) {
+        val entry = manifestDao.value.getManifest(url)
+
+        if (entry == null) {
+            return@withContext null
+        }
+
+        // ignore entry if it hasn't been used for more than 30 days
+        if (!allowTimedout && (System.currentTimeMillis() - entry.usedAt) > usedAtTimeout) {
```

Comment:
Dealing with current time will be a bit of a pain in tests. Maybe we can add the current time as a function parameter.

```kt
suspend fun loadManifest(url: String, includeTimedOut: Boolean = false, currentTime: Long = System.currentTimeMillis())
```